Upstreamize API guide (#3212)
theforeman/foreman-documentation: guides/doc-Project_API/topics/con_api-cheat-sheet.adoc

[id="api-cheat-sheet"]
= API cheat sheet

You can review the following examples of how to use the {ProjectName} API to perform various tasks.

> You can use the API on {ProjectServer} via HTTPS on port 443 or on {SmartProxyServer} via HTTPS on port 8443.

Reviewer: Needs technical ACK please.

Author: API calls through SmartProxy:8443 are deprecated from Foreman 3.12 (checked with SMEs). However, that will require larger changes than just this line...

Author: Or should we just add a deprecation note for now and deal with the rest once it's removed?

You can address these different port requirements within the script itself.

> For example, in Ruby, you can specify the {Project} and {SmartProxy} URLs as follows:

Reviewer: ```suggestion
For example, in Ruby, you can specify the {ProjectServer} and {SmartProxyServer} URLs as follows:
```

[options="nowrap", subs="+quotes,verbatim,attributes"]
----
url = 'https://_{foreman-example-com}_/api/v2/'
{smart-proxy-context}_url = 'https://_{smartproxy-example-com}_:8443/api/v2/'

> katello_url = 'https://_{foreman-example-com}_/katello/api/v2/'

Reviewer: ```suggestion
ifdef::katello,orcharhino,satellite[]
katello_url = 'https://_{foreman-example-com}_/katello/api/v2/'
endif::[]
```

----

> For the host that is subscribed to {ProjectServer} or {SmartProxyServer}, you can determine the correct port required to access the API from the */etc/rhsm/rhsm.conf* file, in the port entry of the `[server]` section.

Reviewer: ```suggestion
For the host that is registered to {ProjectServer} or {SmartProxyServer}, you can determine the correct port required to access the API from the `/etc/rhsm/rhsm.conf` file, in the port entry of the `[server]` section.
```

Reviewer: personal note: I want to check if this is true for all Client OS; tested on AlmaLinux, Ubuntu, and SLES; file belongs to subscription-manager. :heavy_check_mark:

You can use these values to fully automate your scripts, removing any need to verify which ports to use.

The following examples use `curl` for sending API requests.
For more information, see xref:calling-the-api-in-curl[].

> Examples in this chapter use the Python `json.tool` module to format the output.

Reviewer: Kinda unrelated. Maybe we can drop this line?
